### DISCUSSION TITLE
Gliner-Spacy for Numbers

### DISCUSSION THREAD

**Discussion by adb57** [theirstory/gliner-spacy#23] [Q&A] [answered]
I have been really enjoying Gliner-Spacy for Zero Shot NER, simply by fine tuning some of the labels I can almost extract everything I need from a sentence! The only problem I am having is that I cannot seem to get it to pull numbers or units effectively.

For instance the final sentence of [this paper](https://journals.sagepub.com/doi/10.1177/08927057231185710), being:
"However, the optimal printing parameter for PLA polymer are: layer thickness = 0.14 mm, nozzle temperature = 220°C, line width = 0.31 mm and printing speed = 50 mm/s."

If I prompt the following labels:
`configs = {"labels": ["property", "material", "number", "unit_of_measurement"]} `

Gliner-spacy correctly identifies all the properties and materials within the sentence. When it comes to the four numbers it does not identify any as a number, instead identifies _**0.14 mm**_ and _**mm**_ as a unit_of_measurement. It is also failing to identify 220°C, 0.31 and 50 mm/s. 

I know I can use some form of Parts of Speech tagging to identify the numbers and units with NLTK, but I was wondering if there were any ways to perform NER for {number} {units} using Gliner-Spacy?

**Comment by wjbmattingly** [comment, marked as answer]
Thanks for your kind words about GliNER spaCy! This is an interesting question. Identifying numbers that follow specific structures, like degrees and measurements, I wouldn't even go the machine learning route. I would use RegEx. With the right pattern, you could realistically capture 95% of occurrences. For numbers, though, I also have a separate package called Number spaCy: https://spacy.io/universe/project/number-spacy . Once you have a RegEx pattern, you can incorporate that into a spaCy pipeline and have the custom component sitting after (or before!) GliNER.

**Reply by adb57** [reply]
Good to know! I will take a look at both and see how they perform, thanks!

**Comment by wjbmattingly** [comment]
No problem!
